include/test/mir/test/doubles/mock_idle_hub.h

```diff
@@ -37,6 +37,7 @@ struct MockIdleHub : mir::scene::IdleHub
         mir::NonBlockingExecutor&,
         time::Duration));
     MOCK_METHOD1(unregister_interest, void(mir::scene::IdleStateObserver const&));
+    MOCK_METHOD0(inhibit_idle, std::shared_ptr<IdleHub::WakeLock>());
```

Comment:
[Non-blocking]

We try to use `MOCK_METHOD` in new code and clean up `MOCK_METHOD<n>` as we go. E.g.

```suggestion
    MOCK_METHOD(std::shared_ptr<IdleHub::WakeLock>, inhibit_idle, ());
```